staging/src/k8s.io/apiserver/pkg/server/options/encryptionconfig/config.go

```diff
@@ -250,14 +255,26 @@ func loadConfig(filepath string) (*apiserverconfig.EncryptionConfiguration, erro
 
 	configObj, gvk, err := codecs.UniversalDecoder().Decode(data, nil, nil)
 	if err != nil {
-		return nil, err
+		return nil, "", 0, err
 	}
 	config, ok := configObj.(*apiserverconfig.EncryptionConfiguration)
 	if !ok {
-		return nil, fmt.Errorf("got unexpected config type: %v", gvk)
+		return nil, "", 0, fmt.Errorf("got unexpected config type: %v", gvk)
 	}
 
-	return config, validation.ValidateEncryptionConfiguration(config).ToAggregate()
+	// compare KMS timeout and determine max timeout
+	var maxTimeout time.Duration
+	for _, resourceConfig := range config.Resources {
+		for _, providerConfig := range resourceConfig.Providers {
+			if providerConfig.KMS != nil {
+				if providerConfig.KMS.Timeout.Duration > maxTimeout {
+					maxTimeout = providerConfig.KMS.Timeout.Duration
+				}
+			}
+		}
+	}
```

Comment:
Move this to

https://github.com/kubernetes/kubernetes/blob/2ab2698e4c33aa0c103ebd0459ccfbef486b7e3c/staging/src/k8s.io/apiserver/pkg/server/options/encryptionconfig/config.go#L285-L289

It is more code to funnel the information up but I prefer all the KMS logic in one place.